docs/python/documentprojection/__main__.py

```diff
@@ -0,0 +1,115 @@
+from utils.logging import *
+from utils.notebook import *
+from framework.pipeline import *
+from channels.website import *
+
+import re
+
+log = get_log(__name__)
+
+
+def get_channel_map():
+    def camel_to_snake(name):
+        s1 = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", name)
+        s2 = re.sub("([a-z0-9])([A-Z])", r"\1_\2", s1).lower()
+        return s2.replace("_channel", "")
+
+    channel_map = {
+        k: v
+        for k, v in [
+            (camel_to_snake(channel.__name__), channel) for channel in all_channels
+        ]
+    }
+    return channel_map
+
+
+def parse_args():
+    channel_choice_names = ["all"] + list(get_channel_map())
+    log_level_choices = ["debug", "info", "warn", "error", "critical"]
+
+    import argparse
+
+    parser = argparse.ArgumentParser(description="Document Projection Pipeline")
+    parser.add_argument(
+        "notebooks",
+        metavar="N",
+        type=str,
+        nargs="*",
+        help="a notebook or folder of notebooks to project",
+    )
+    parser.add_argument(
+        "-t", "--test", action="store_true", default=False, help="run only tests"
+    )
```

Comment:
Does this system need to run tests? Right now thats handled by scala so you might not need to worry about this